### ISSUE TITLE
Adding ASCII ART Text builder (For inPageBuilder)

### ISSUE THREAD

**Issue by Elius94** [Elius94/console-gui-tools#55]

**Comment by h-sifat** [comment]
Hi Elius, is there anything that I can help you with?

**Comment by Elius94** [comment]
Hello! Nice to hear that! If you want, of course you can help me to solve some issues. Since this is a new feature, you can propose this add-on wit a pull request, so I'll review and if it's all ok, I will release it.

In this particular case: it could be difficult because to make a generator of ASCII art you need to have a font archive and I think this can take up a lot of space. You could take a cue from an open source library that already has this thing. However, you may need to make it an installable plugin from a second package .however, if you want to help me, keep in mind that:
 - Must be a function that returns an Array of StyledElements 
 - It must be able to be added to a common PageBuilder.
 - No other files should be modified. Only fresh files should suffice.

Thank you very much for helping.
If I think it is a valid proposal, I will be happy to release your contribution.

**Comment by Elius94** [comment]
Hello! Nice to hear that! If you want, of course you can help me to solve some issues. Since this is a new feature, you can propose this add-on wit a pull request, so I'll review and if it's all ok, I will release it.

In this particular case: it could be difficult because to make a generator of ASCII art you need to have a font archive and I think this can take up a lot of space. You could take a cue from an open source library that already has this thing. However, you may need to make it an installable plugin from a second package .however, if you want to help me, keep in mind that:
 - Must be a function that returns an Array of StyledElements 
 - It must be able to be added to a common PageBuilder.
 - No other files should be modified. Only fresh files should suffice.

Thank you very much for helping.
If I think it is a valid proposal, I will be happy to release your contribution.

**Comment by h-sifat** [comment]
Thanks for the explanation. After my frustration with **Ink** and **Blessed** I've just found your library today and it seems very promising to me. Though, I haven't used it yet but I would love to be part of it. 

I actually didn't mean to help with this specific feature because I don't know much about it (but I'm willing to learn it if needed). If you don't mind, could you kindly give me a beginner friendly guideline about how I can get started with this project. Is there any prerequisites (other than Node.js and Typescript) that I should have before studying the source code?

Thank you 💝.

**Comment by Elius94** [comment]
> Thanks for the explanation. After my frustration with **Ink** and **Blessed** I've just found your library today and it seems very promising to me. Though, I haven't used it yet but I would love to be part of it.
> 
> I actually didn't mean to help with this specific feature because I don't know much about it (but I'm willing to learn it if needed). If you don't mind, could you kindly give me a beginner friendly guideline about how I can get started with this project. Is there any prerequisites (other than Node.js and Typescript) that I should have before studying the source code?
> 
> Thank you 💝.

Thanks a lot for the interest!
To start with something easier, you can focus on this other issue:
https://github.com/Elius94/console-gui-tools/issues/54

It should be a quick work to add this feature to the progress bar class. 
So, if you upload a PR, I will happy to check it and if it's all fine, I will merge it :)

**Comment by Elius94** [comment]
> Is there any prerequisites (other than Node.js and Typescript) that I should have before studying the source code?

No, you only need Node (14 or higher) and all the developer dependency (see in Package.json)

 1. Clone the repo or fork it.
 2. Open your text editor inside.
 3. Edit.
 4. Run `npm run build`
 5. Make a new example file in the examples folder and make it works with your code and your patch
 6. Once you have checked that all is fine, make a pull request directly from your fork.

Finally I'll check your mod by myself.
